File: payola/web/public/javascripts/cz/payola/web/client/views/Layer.js
goog.provide('cz.payola.web.client.views.Layer');
goog.require('cz.payola.web.client.views.Vector');
cz.payola.web.client.views.Layer = function(canvas, context) {
var self = this;
self.canvas = canvas;
self.context = context;
self.cleared = true;
};
cz.payola.web.client.views.Layer.prototype.setSize = function(size) {
var self = this;
self.canvas.width = size.x;
self.canvas.height = size.y;
};
cz.payola.web.client.views.Layer.prototype.getSize = function() {
var self = this;
return new cz.payola.web.client.views.Vector(self.canvas.width, self.canvas.height);
};
cz.payola.web.client.views.Layer.prototype.metaClass_ = new s2js.MetaClass('cz.payola.web.client.views.Layer', []);

File: payola/web/client/src/main/scala/cz/payola/web/client/views/plugins/visual/Layer.scala
package cz.payola.web.client.views.plugins.visual

import s2js.adapters.js.dom.{CanvasRenderingContext2D, Canvas}

/**
  * Representation of a space for drawing into a web page.
  * @param canvas object specifying sizes of the context
  * @param context object for drawing
  */
class Layer(val canvas: Canvas, val context: CanvasRenderingContext2D)
{
    /**
      * Indicator whether is the layer ready for drawing into.
      */
    var cleared = false

    /**
      * Setter of canvas dimensions.
      * @param size new dimensions.
      */
    def setSize(size: Vector) {
        canvas.width = size.x;
        canvas.height = size.y;
    }

    /**
      * Getter of canvas dimensions.
      * @return dimensions
      */
    def getSize: Vector = {
        Vector(canvas.width, canvas.height)
    }
}
